src/Core/Config/ConfigCache.php

```diff
@@ -0,0 +1,183 @@
+<?php
+
+namespace Friendica\Core\Config;
+
+/**
+ * The Friendica config cache for the application
+ * Initial, all *.config.php files are loaded into this cache with the
+ * ConfigCacheLoader ( @see ConfigCacheLoader )
+ *
+ * Is used for further caching operations too (depending on the ConfigAdapter )
+ */
+class ConfigCache implements IConfigCache, IPConfigCache
+{
+	/**
+	 * NEVER, EVER use this public config array outside of the class
+	 * It is only public due to backward compatibility to .htconfig.php
+	 *
+	 * @var array The cached config array
+	 */
+	public $config;
+
+	/**
+	 * @param array $config    A initial config array
+	 */
+	public function __construct($config = [])
+	{
+		$this->config = [];
+
+		if (isset($config)) {
+			$this->loadConfigArray($config, true);
+		}
+	}
+
+	/**
+	 * Tries to load the specified configuration array into the App->config array.
+	 * Doesn't overwrite previously set values by default to prevent default config files to supersede DB Config.
+	 *
+	 * @param array $config
+	 * @param bool  $overwrite Force value overwrite if the config key already exists
+	 */
+	public function loadConfigArray(array $config, $overwrite = false)
+	{
+		foreach ($config as $category => $values) {
+			foreach ($values as $key => $value) {
+				if ($overwrite) {
+					self::set($category, $key, $value);
+				} else {
+					self::setDefault($category, $key, $value);
+				}
+			}
+		}
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function get($cat, $key = null, $default = null)
+	{
+		$return = $default;
+
+		if ($cat === 'config') {
+			if (isset($this->config[$key])) {
+				$return = $this->config[$key];
+			}
+		} else {
+			if (isset($this->config[$cat][$key])) {
+				$return = $this->config[$cat][$key];
+			} elseif ($key == null && isset($this->config[$cat])) {
+				$return = $this->config[$cat];
+			}
+		}
+
+		return $return;
+	}
+
+	/**
+	 * Sets a default value in the config cache. Ignores already existing keys.
+	 *
+	 * @param string $cat Config category
+	 * @param string $k   Config key
+	 * @param mixed  $v   Default value to set
+	 */
+	private function setDefault($cat, $k, $v)
+	{
+		if (!isset($this->config[$cat][$k])) {
+			self::set($cat, $k, $v);
```

Comment:
Static/Object confusion, should be `$this->`.
```suggestion
			$this->set($cat, $k, $v);
```